server/instance.go

```diff
@@ -488,22 +491,25 @@ func (s *Server) createInstance(ctx context.Context, create *store.InstanceCreat
 	// Try creating the "bytebase" db in the added instance if needed.
 	// Since we allow user to add new instance upfront even providing the incorrect username/password,
 	// thus it's OK if it fails. Frontend will surface relevant info suggesting the "bytebase" db hasn't created yet.
-	db, err := s.dbFactory.GetAdminDatabaseDriver(ctx, instance, "" /* databaseName */)
+	driver, err := s.dbFactory.GetAdminDatabaseDriver(ctx, instance, "" /* databaseName */)
 	if err == nil {
-		defer db.Close(ctx)
-		if err := db.SetupMigrationIfNeeded(ctx); err != nil {
+		defer driver.Close(ctx)
+		if err := driver.SetupMigrationIfNeeded(ctx); err != nil {
 			log.Warn("Failed to setup migration schema on instance creation",
 				zap.String("instance_name", instance.Name),
 				zap.String("engine", string(instance.Engine)),
 				zap.Error(err))
 		}
-		if _, err := s.SchemaSyncer.SyncInstance(ctx, instance); err != nil {
-			log.Warn("Failed to sync instance",
-				zap.Int("instance_id", instance.ID),
-				zap.Error(err))
+		// TODO(zp): support sync instance and database for MongoDB.
+		if instance.Engine != db.MongoDB {
```

Comment:
We also have other places to call the `SchemaSyncer.SyncInstance` like instance update. Maybe take the check in the `SyncInstance` func?

Reply:
SGTM